### PR TITLE
Verify remaining test projects without Startup class

### PR DESCRIPTION
**Pull request by Saibamen**
And also test them in `SwaggerIntegrationTests.cs`

### CODE REVIEW

File: test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs [hunk 1]
```diff
@@ -162,11 +162,11 @@ public async Task TypesAreRenderedCorrectly()
             ]);
         }
 
-        private static async Task SwaggerEndpointReturnsValidSwaggerJson<TEntryPoint>(string swaggerRequestUri)
-            where TEntryPoint : class
+        private static async Task SwaggerEndpointReturnsValidSwaggerJson(Type entryPointType, string swaggerRequestUri)
         {
-            using var application = new TestApplication<TEntryPoint>();
-            using var client = application.CreateDefaultClient();
+            using var application = (dynamic)Activator.CreateInstance(typeof(TestApplication<>).MakeGenericType(entryPointType));
```
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs:167]: Please refactor to not use `dynamic`.

File: test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerVerifyIntegrationTest.cs [hunk 2]
```diff
@@ -92,11 +91,11 @@ public async Task TypesAreRenderedCorrectly()
             await Verifier.Verify(swaggerResponse);
         }
 
-        private static async Task<string> SwaggerEndpointReturnsValidSwaggerJson<TEntryPoint>(string swaggerRequestUri)
-            where TEntryPoint : class
+        private static async Task<string> SwaggerEndpointReturnsValidSwaggerJson(Type entryPointType, string swaggerRequestUri)
         {
-            using var application = new TestApplication<TEntryPoint>();
-            using var client = application.CreateDefaultClient();
+            using var application = (dynamic)Activator.CreateInstance(typeof(TestApplication<>).MakeGenericType(entryPointType));
```
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerVerifyIntegrationTest.cs:96]: +1

File: test/WebSites/MinimalAppWithHostedServices/Program.cs [hunk 3]
```diff
@@ -24,11 +24,21 @@ class HostedService : IHostedService
 {
     public Task StartAsync(CancellationToken cancellationToken)
     {
-        throw new Exception("Crash!");
```
Comment (martincostello) [inline test/WebSites/MinimalAppWithHostedServices/Program.cs:27]: Was this here to test something specific?

File: test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs [hunk 4]
```diff
@@ -21,6 +21,8 @@ public class SwaggerIntegrationTests
         [Theory]
         [InlineData(typeof(Basic.Startup), "/swagger/v1/swagger.json")]
         [InlineData(typeof(CliExample.Startup), "/swagger/v1/swagger_net8.0.json")]
+        // TODO: FIXME: CliExampleWithFactory project is not working
+        //[InlineData(typeof(CliExampleWithFactory.Startup), "/swagger/v1/swagger_net8.0.json")]
```
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs:25]: I'd rather these TODOs are fixed in the same PR _or_ a separate PR is merged first that fixes them and removes the need for the TODOs.

Comment (Saibamen) [reply test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs]: Fixed

File: test/WebSites/MinimalAppWithHostedServices/Program.cs (hunk 3 above)
Comment (martincostello) [reply test/WebSites/MinimalAppWithHostedServices/Program.cs:27]: It was added intentionally (by me in turns out) as part of #2880 to verify that _other_ `IHostedService` aren't invoked by our CLI tool.

_Either_ that coverage needs to be retained somewhere else, or this project should be re-excluded from the Verification tests.

Comment (Saibamen) [reply test/WebSites/MinimalAppWithHostedServices/Program.cs:27]: Done. Added comment in Program.cs about this

File: Directory.Packages.props [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <Project>
   <ItemGroup>
-    <PackageVersion Include="Autofac.Extensions.DependencyInjection" Version="4.2.2" />
+    <PackageVersion Include="Autofac.Extensions.DependencyInjection" Version="7.2.0" />
```
Comment (Saibamen) [inline Directory.Packages.props:3]: This is the minimum version working with .NET 6 - no startup exceptions at all.

But should I update it to version 10 (also works without any code changes)?

Comment (martincostello) [reply Directory.Packages.props:3]: If you're confident the tests will find any breakages, upgrading it to the latest is fine.

File: test/WebSites/MinimalAppWithHostedServices/Program.cs [hunk 6]
```diff
@@ -24,6 +24,7 @@ class HostedService : IHostedService
 {
     public Task StartAsync(CancellationToken cancellationToken)
     {
+        // This is intended and should be not tested in this implementation! See https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/3151#discussion_r1856678972
```
Comment (martincostello) [inline test/WebSites/MinimalAppWithHostedServices/Program.cs:27]:
```suggestion
        // This is intentional. See https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/3151#discussion_r1856678972
```

**Review by martincostello — APPROVED** [review]

**Pull request merged**